### ISSUE TITLE
Compiling a Cmm file

### ISSUE THREAD

**Issue by kavon** [kavon/ghc#2]
The `codeGen/should_compile` tests `T10518` and  `T9329` run into the same issue:

```
=====> T10518(llvm) 25 of 28 [0, 0, 0]
cd "./T10518.run" &&  "/Users/kavon/msr/ghc/inplace/test   spaces/ghc-stage2" -c T10518.cmm -dcore-lint -dcmm-lint -no-user-package-db -rtsopts -fno-warn-missed-specialisations -fshow-warning-groups -fdiagnostics-color=never -fno-diagnostics-show-caret -dno-debug-output  -fllvm  -no-hs-main 
Compile failed (exit code 1) errors were:
ghc-stage2: panic! (the 'impossible' happened)
  (GHC version 8.3.20170906 for x86_64-apple-darwin):
	runPhase: don't know how to run phase: LlvmOpt

Please report this as a GHC bug:  http://www.haskell.org/ghc/reportabug
```

It seems that compiling those `.cmm` files is broken right now, due to the change in the driver pipeline to carry mangler info.

**Comment by kavon** [comment]
Also `T7571` and `T9050` fails due to this.

**Comment by kavon** [comment]
Fixed in 4790509e7b52f5904d50ac937998b47b347fedac
